corgi/core/models.py

```diff
@@ -384,36 +382,8 @@ def components(self) -> "ComponentQuerySet":
 
     @property
     def builds(self) -> QuerySet:
-        """Returns unique productcomponentrelations with at least 1 matching variant or stream,
-        ordered by build_id.
-        """
-        # TODO: Can just return self.components.values_list("software_build", flat=True).distinct()
-        product_refs = [self.name]
-        if isinstance(self, ProductStream):
-            # we also want to include child product variants of this product stream
-            product_refs.extend(
-                self.productvariants.values_list("name", flat=True).using("read_only")
-            )
-        elif isinstance(self, ProductVersion) or isinstance(self, Product):
-            # we don't store products or product versions in the relations table therefore
-            # we only want to include child product streams and product variants in the query
-            product_refs.extend(
-                self.productstreams.values_list("name", flat=True).using("read_only")
-            )
-            product_refs.extend(
-                self.productvariants.values_list("name", flat=True).using("read_only")
-            )
-        # else it was a product variant, only look up by self.name
-
-        if product_refs:
-            return (
-                ProductComponentRelation.objects.filter(product_ref__in=product_refs)
-                .values_list("build_id", "build_type")
-                .distinct()
-                .using("read_only")
-            )
-        # Else no product variants or product streams - should never happen
-        return ProductComponentRelation.objects.none()
+        """Returns distinct software builds associated with this Product instance via components"""
+        return self.components.values_list("software_build", flat=True).distinct()
```

Comment:
Querying for builds this way is really slow, now that we have a foreign key between SoftwareBuild and ProductComponentRelation I think it might actually be better to revert this to the old way, but just add a filter excluding relations where software_build is None.

This builds function is only used to get a build count in the API and tests today. However it's often a useful function to use in scripts, and probably more useful if it returned SoftwareBuild objects rather than Relations.